lib/task/import/validate/csvCultureValidator.class.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV culture column test. Check if present, check values against master list,
+ * and check if piped value.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ *
+ * @internal
+ * @coversNothing
+ */
+class CsvCultureValidator extends CsvBaseValidator
+{
+    const TITLE = 'Culture Check';
+    protected $cultureColumnPresent;
+    protected $rowsWithBlankCulture = 0;
+    protected $rowsWithPipeFoundInCulture = 0;
+    protected $rowsWithInvalidCulture = 0;
+    protected $invalidCultures = [];
+
+    public function __construct(array $options = null)
+    {
+        $this->setTitle(self::TITLE);
+
+        parent::__construct($options);
+    }
+
+    public function reset()
+    {
+        $this->cultureColumnPresent = null;
+        $this->rowsWithBlankCulture = 0;
+        $this->rowsWithPipeFoundInCulture = 0;
+        $this->rowsWithInvalidCulture = 0;
+        $this->invalidCultures = [];
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+        $row = $this->combineRow($header, $row);
+
+        // Set if culture column is present.
+        if (!isset($this->cultureColumnPresent)) {
+            $this->cultureColumnPresent = array_key_exists('culture', $row);
+        }
+
+        // If present check contents.
+        if ($this->cultureColumnPresent) {
+            if (empty($row['culture'])) {
+                ++$this->rowsWithBlankCulture;
+            } else {
+                // Check if contains pipe.
+                if (0 < strpos($row['culture'], '|')) {
+                    ++$this->rowsWithPipeFoundInCulture;
+                    $this->testData->addDetail(implode(',', $row));
+
+                    // Keep a list of invalid culture values.
```

Comment:
I would restructure this if/elseif to avoid repeating this code block twice. e.g.

if ($this->isCultureValid($row['culture'])) {
    return;
}

... check for pipe or invalid culture string

Reply:
Another great suggestion. Change completed.